Blast xml fix

kescobo opened the pull request:
Per request from @bicycle1885 - I altered the BLAST results parser to use EzXML rather than LightXML. This works, though I'm not sure that using XPath queries all over the place is the most efficient approach. Happy to incorporate any feedback if there's a better way.

I added EzXML to REQUIRE, but did not remove LightXML, since I think there's something else that still relies on it.

bicycle1885 commented on src/tools/blast/blastcommandline.jl:
```diff
@@ -26,32 +26,29 @@ readblastXML(results)
+function readblastXML(blastrun::String; seqtype="nucl")
+    dc = parsexml(blastrun)
+    rt = root(dc)
+    results = Vector{BLASTResult}([])
```
`results = BLASTResult[]` would be a conventional syntax.

bicycle1885 commented on src/tools/blast/blastcommandline.jl:
```diff
@@ -26,32 +26,29 @@ readblastXML(results)
+    dc = parsexml(blastrun)
+    rt = root(dc)
+    results = Vector{BLASTResult}([])
+    for iteration in find(rt, "//BlastOutput_iterations/Iteration")
```
`find(rt, "/BlastOutput/BlastOutput_iterations/Iteration")` is more explicit and reduces the search space.

bicycle1885 commented on src/tools/blast/blastcommandline.jl:
```diff
@@ -26,32 +26,29 @@ readblastXML(results)
+    rt = root(dc)
+    results = Vector{BLASTResult}([])
+    for iteration in find(rt, "//BlastOutput_iterations/Iteration")
+        queryname = content(findfirst(iteration, ".//Iteration_query-def"))
```
You can write it as `findfirst(iteration, "Iteration_query-def")`.

bicycle1885 commented:
BTW, I couldn't find code that depends on LightXML except BLAST tools. Does dropping LightXML from dependencies cause TravisCI failure or something?

kescobo commented:
@bicycle1885 Thanks for the comments. I think I addressed them all in the latest commit. 

re: search space for things I'm hitting with `.//blah` type searches - the results are explicitly structured enough that I could probably eliminate all of them. Is it better if I'm explicit as possible? I'm probably just being lazy typing...

re: LightXML - Looking back, I think I was thinking of something that you were developing that you said you started using LightXML for, but it makes sense you switched to EzXML already. I removed it from REQUIRE.

bicycle1885 commented on src/tools/blast/blastcommandline.jl:
```diff
@@ -16,7 +16,7 @@ immutable BLASTResult
 
 """
-    readblastXML(blastrun::AbstractString)
+    readblastXML(blastrun::String)
```
Could you restore the original argument type here (and `readblastXML`)?

kescobo replied:
Oops - sorry, thought I'd caught them all. Fixed!

Ward9250 commented:
I was also experimenting with LightXML in the past, so that may well be why it is in there too.

Ward9250 commented:
@kescobo Nah I don't think so right now, and when it comes to needing XML again, I'll just use Ez :)

bicycle1885 commented:
Looks good 👍  I'll merge soon.
